content/en/blog/_posts/2019-12-09-csi-volume-migration-beta.md

```diff
@@ -0,0 +1,149 @@
+What is the timeline/status?
+The timeline for CSI migration is actually set by the cloud provider extraction project. It is part of the effort to remove all cloud provider code from Kubernetes. By migrating cloud storage plugins to external CSI drivers we are able to extract out all the cloud provider dependencies. Since an official deprecation was announced in TODO we plan to follow the timeline and move CSI migration to GA and completely remove all in-tree plugin code from Kubernetes by v1.21.
+
+Although the overall feature is beta and not on by default, there is still work to be done on a per-plugin basis. Currently only GCE PD and AWS EBS have gone beta with Migration and yet both are still off by default since they depend on a manual installation of their respective CSI Drivers. Azure File/Disk, OpenStack, and VMWare plugins are currently in less mature states and non-cloud plugins such as NFS, Portworx, RBD etc are still in the planning stages.
+
+The current and targeted releases for each individual cloud driver is shown in the table below:
+
+| Driver            | Alpha         | Beta (in-tree deprecated) | GA            | Target "in-tree plugin" removal |
+| ----------------- | ------------- | ------------------------- | ------------- | ------------------------------- |
+| AWS EBS           | 1.14          | 1.17                      | 1.19 (Target) | 1.21                            |
+| GCE PD            | 1.14          | 1.17                      | 1.19 (Target) | 1.21                            |
+| OpenStack Cinder  | 1.14          | 1.18 (Target)             | 1.19 (Target) | 1.21                            |
+| Azure Disk + File | 1.15          | 1.18 (Target)             | 1.19 (Target) | 1.21                            |
+| VSphere           | 1.18 (Target) | 1.19 (Target)             | 1.20 (Target) | 1.22                            |
+
+## What's next?
+
+Major upcoming work includes implementing and hardening CSI migration for the remaining in-tree plugins, installing CSI Drivers by default in distributions, turning on CSI migration by default, and finally removing all in-tree plugin code as a part of cloud provider extraction. We expect to complete this project including the full switch to “on-by-default” migration by Kubernetes v1.21.
+
+## What should I do as a user?
+
+Please note that all new features for the Kubernetes storage system (like volume snapshotting) will only be added to the CSI interface. Therefore, if you are starting up a new cluster, creating stateful applications for the first time, or require these new features we recommend using CSI drivers natively (instead of the in-tree volume plugin API). Please follow the updated user guides and use the new CSI APIs.
```

Comment:
Optional rewording:
```suggestion
Note that all new features for the Kubernetes storage system (like volume snapshotting) will only be added to the CSI interface. Therefore, if you are starting up a new cluster, creating stateful applications for the first time, or require these new features we recommend using CSI drivers natively (instead of the in-tree volume plugin API). Follow the updated user guides to use the new CSI APIs.
```
(rewording to _avoid_ please, as per style guidelines)